zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/EagerACLFilterTest.java

```
@@ -0,0 +1,297 @@
+    protected final static byte[] AUTH = "hello".getBytes();
+    protected final static byte[] AUTHB = "goodbye".getBytes();
+    protected final static byte[] DATA = "Hint Water".getBytes();
+    protected TestableZooKeeper zkClient;
+    protected TestableZooKeeper zkClientB;
+    protected QuorumPeer zkLeader;
+    protected ZooKeeperServer connectedServer;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(
+                new Object[][] {
+                        {ServerState.LEADING, true},
+                        {ServerState.LEADING, false},
+                        {ServerState.FOLLOWING, true},
+                        {ServerState.FOLLOWING, false},
+                        {ServerState.OBSERVING, true},
+                        {ServerState.OBSERVING, false}});
+    }
+
+    public EagerACLFilterTest(ServerState state, boolean checkEnabled) {
+        this.serverState = state;
+        this.checkEnabled = checkEnabled;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        ensureCheck(checkEnabled);
+        CountdownWatcher clientWatch = new CountdownWatcher();
+        CountdownWatcher clientWatchB = new CountdownWatcher();
+        super.setUp(true);
+
+        String hostPort = getPeersMatching(serverState).split(",")[0];
+        int clientPort = Integer.parseInt(hostPort.split(":")[1]);
+
+        zkLeader = getPeerList().get(getLeaderIndex());
+        connectedServer = getPeerByClientPort(clientPort).getActiveServer();
+
+        zkClient = createClient(clientWatch, hostPort);
+        zkClientB = createClient(clientWatchB, hostPort);
+        zkClient.addAuthInfo(AUTH_PROVIDER, AUTH);
+        zkClientB.addAuthInfo(AUTH_PROVIDER, AUTHB);
+        clientWatch.waitForConnected(CONNECTION_TIMEOUT);
+        clientWatchB.waitForConnected(CONNECTION_TIMEOUT);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        zkClient.close();
```

Please add null check for the case when `setUp()` fails early.

---

done.